### PR TITLE
Add `render_markdown` parameter to chatbot

### PR DESCRIPTION
**Pull request by dawoodkhan82**
## Description
Allow users to `disable_markdown` for chatbot messages.

Closes: #5513

## 🎯 PRs Should Target Issues

Before your create a PR, please check to see if there is [an existing issue](https://github.com/gradio-app/gradio/issues) for this change. If not, please create an issue before you create this PR, unless the fix is very small. 

Not adhering to this guideline will result in the PR being closed. 

## Tests

1. PRs will only be merged if tests pass on CI. To run the tests locally, please set up [your Gradio environment locally](https://github.com/gradio-app/gradio/blob/main/CONTRIBUTING.md) and run the tests: `bash scripts/run_all_tests.sh`

2. You may need to run the linters: `bash scripts/format_backend.sh` and `bash scripts/format_frontend.sh`

### CODE REVIEW

File: gradio/components/chatbot.py [hunk 1]
```diff
@@ -76,6 +77,7 @@ def __init__(
             elem_classes: An optional list of strings that are assigned as the classes of this component in the HTML DOM. Can be used for targeting CSS styles.
             height: height of the component in pixels.
             latex_delimiters: A list of dicts of the form {"left": open delimiter (str), "right": close delimiter (str), "display": whether to display in newline (bool)} that will be used to render LaTeX expressions. If not provided, `latex_delimiters` is set to `[{ "left": "$$", "right": "$$", "display": True }]`, so only expressions enclosed in $$ delimiters will be rendered as LaTeX, and in a new line. Pass in an empty list to disable LaTeX rendering. For more information, see the [KaTeX documentation](https://katex.org/docs/autorender.html).
+            disable_markdown: If True, will disable Markdown rendering for chatbot messages.
```
Comment (abidlabs) [inline gradio/components/chatbot.py:80]: tiniest nit: you could consider calling it `render_markdown` and setting it to True by default to be more parallel with `sanitize_html`

File: js/markdown/static/MarkdownCode.svelte [hunk 2]
```diff
@@ -24,10 +25,19 @@
...
+
+	function process_message(value: string): string {
+		if (sanitize_html && !disable_markdown) {
+			return DOMPurify.sanitize(marked.parse(value));
+		} else if (sanitize_html && disable_markdown) {
+			return DOMPurify.sanitize(value);
+		} else if (!sanitize_html && !disable_markdown) {
+			return marked.parse(value);
+		}
+		return value;
```
Comment (abidlabs) [inline js/markdown/static/MarkdownCode.svelte:37]: You could simplify this by doing something like:
```
if (!disable_markdown): {
   value = marked.parse(value)
}
if (sanitizer_html) {
   value = DOMPurify.sanitize(value)
}
return value;
```

**Comment by abidlabs** [comment]
@dawoodkhan82 I think there's still one change that needs to be made -- if Markdown is disabled, then we should be rendering the string the user as a regular string, instead of as html.

I tested this example:

```py
import gradio as gr

long_msg = """
Hi **Test**

one1 <Color2>

<Color3> two2
<Color4> three3
"""

with gr.Blocks() as demo:
    gr.Chatbot(
        [[long_msg, long_msg]],
        height=550,
        disable_markdown=True,
        sanitize_html=False,
    )
demo.launch()
```

You'll notice that the `<Color>` tags don't appear:

image

I believe that's because we're still rendering the content as html instead of as a regular string:

https://github.com/gradio-app/gradio/blob/1bc8471abc31dd4a7323b7275971dbcaf4f8fdd2/js/markdown/static/MarkdownCode.svelte#L56

**Comment by dawoodkhan82** [comment]
Thanks for the review, and catching the html bug. I made all the changes

**Comment by abidlabs** [comment]
Looks good @dawoodkhan82 -- one last thing I noticed is that if `render_markdown` is False, then new lines are not preserved. So the message ends up looking like this:

image

We should be able to fix this by adding `white-space: pre-line;` to the style if `render_markdown` is False.

**Comment by dawoodkhan82** [comment]
@abidlabs fixed the new line issue

**Comment by abidlabs** [comment]
LGTM thanks for addressing everything @dawoodkhan82! I'll merge this in to reduce the number of open PRs :)

**Pull request merged**